libbeat/monitoring/inputmon/httphandler.go

```diff
@@ -73,10 +73,17 @@ func (h *handler) allInputs(w http.ResponseWriter, req *http.Request) {
 			continue
 		}
 
-		if requestedType != "" {
-			if typ, ok := m["input"].(string); ok && !strings.EqualFold(typ, requestedType) {
-				continue
-			}
+		// Require all entries to have an 'input' and 'id' to be accessed through this API.
+		inputType, ok := m["input"].(string)
+		if !ok {
+			continue
+		}
+		if _, ok = m["id"].(string); !ok {
+			continue
+		}
+
+		if requestedType != "" && !strings.EqualFold(inputType, requestedType) {
+			continue
 		}
```

Comment:
```suggestion
		if _, ok = m["id"].(string); !ok {
			continue
		}
		
		if inputType, ok := m["input"].(string); !ok || (requestedType != "" && !strings.EqualFold(inputType, requestedType)) {
			continue
		}
```
(semantic bundling at the cost of line length ­— stylistic suggestion only, would also suggest s/inputType/typ/g, but with even less weight)